File: frontend/src/components/System/Criticality/Criticality.js
import React from 'react'
import './Criticality.css'

const Criticality = (props) => {
    switch (props.level) {
        case 'cni':
            return <span className="badge criticality-cni">CNI</span>
        case 'high':
            return <span className="badge criticality-high">HIGH</span>
        case 'medium':
            return <span className="badge criticality-medium">MEDIUM</span>
        case 'low':
            return <span className="badge criticality-low">LOW</span>
        default:
            return <span className="badge criticality-unknown">UNKNOWN</span>
    }
}


export default Criticality

File: backend/src/main/java/uk/gov/digital/ho/systemregister/application/messaging/commands/validation/Criticality.java
package uk.gov.digital.ho.systemregister.application.messaging.commands.validation;

import javax.validation.Constraint;
import javax.validation.Payload;
import javax.validation.constraints.Pattern;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

//TODO update to custom validator with dynamic values in default message
@Constraint(validatedBy = {})
@Pattern(regexp = "high|low|medium|cni|unknown", message = "Criticality must be one of the following values: high, low, medium, cni, unknown")
@Retention(RUNTIME)
@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE})
@SuppressWarnings("unused")
public @interface Criticality {
    String message() default "";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}
